### PR TITLE
Support inline val references in top level splices

### LINKED ISSUE #12196: inline val/defs can't be used when calling a quotes fn
## Compiler version

This used to work in RC1. It fails in RC2 and RC3.

## Minimized code

```Scala
import scala.quoted.*

def qqq(s: String)(using Quotes): Expr[Unit] = '{()}

inline val InlineVal = "i"
inline def InlineDef = "i"

inline def withInlineVal = ${ qqq(InlineVal) }
inline def withInlineDef = ${ qqq(InlineDef) }
inline def withString    = ${ qqq("i") }
```

## Output

```
[error] -- Error: x.scala:8:34 --------------
[error] 8 |inline def withInlineVal = ${ qqq(InlineVal) }
[error]   |                                  ^^^^^^^^^
[error]   |                                  Malformed macro parameter
[error]   |
[error]   |                                  Parameters may only be:
[error]   |                                   * Quoted parameters or fields
[error]   |                                   * Literal values of primitive types
[error] -- Error: x.scala:9:34 --------------
[error] 9 |inline def withInlineDef = ${ qqq(InlineDef) }
[error]   |                                  ^^^^^^^^^
[error]   |                                  Malformed macro parameter
[error]   |
[error]   |                                  Parameters may only be:
[error]   |                                   * Quoted parameters or fields
[error]   |                                   * Literal values of primitive types
[error] two errors found
```

## Expectation

Compilation.

### PR DESCRIPTION
**Pull request by nicolasstucki**
Fixes #12196

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/transform/Splicer.scala [hunk 1]
```diff
@@ -242,6 +243,11 @@ object Splicer {
       case Literal(Constant(value)) =>
         interpretLiteral(value)
 
+      case tree: Ident if tree.symbol.is(Inline, butNot = Method) =>
+        tree.tpe.widenTermRefExpr match
+          case ConstantType(c) => c.value.asInstanceOf[Object]
```
Comment (japgolly) [inline compiler/src/dotty/tools/dotc/transform/Splicer.scala:248]: We should probably handle `null` here too

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/transform/Splicer.scala:248]: We do not support `inline val n = null`. That should be added when we add support for it and can write a test case for that logic.

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/transform/Splicer.scala:248]: See #12198

**Review by liufengyun — APPROVED** [review]
LGTM

**Pull request merged**